### PR TITLE
parcoords - bug fix handling undefined ranges and dticks for tickText

### PR DESCRIPTION
**Pull request by archmoj**
In addition to some refactors (e.g. to avoid using one variable name in two scopes), 
this PR fixes the `tickText` defaults for undefined ranges and ticks to be used by `tickText`.

Please note the changes in two baselines would make them look similar to what they look before #3966. 
@plotly/plotly_js

### CODE REVIEW

File: src/traces/parcoords/defaults.js [hunk 1]
```diff
@@ -55,6 +55,19 @@ function dimensionDefaults(dimensionIn, dimensionOut, parentOut, opts) {
         coerce('ticktext');
         coerce('tickformat');
         var range = coerce('range');
+        if(!range) {
```
Comment (etpinard) [inline src/traces/parcoords/defaults.js]: Hmm I would move this down to the plot step. That way we don't have to loop over the data during the defaults.

**Comment by etpinard** [comment]
Nice fix @archmoj - I made one small comment.

File: src/traces/parcoords/defaults.js (hunk 1 above)
Comment (archmoj) [reply src/traces/parcoords/defaults.js]: Good call.
Done in https://github.com/plotly/plotly.js/pull/4011/commits/bf02101c5d8c51e2db07065f1a7ab20ce0e8ff55.

**Comment by etpinard** [comment]
Excellent. :dancer: 

Sorry about those intermittent test failures. I'm working on it at the moment.

**Pull request merged**